pkg/orchestration/wiring/internaldns/plugin.go

```diff
@@ -131,21 +119,13 @@ func getReferences(resource *orch_v1.StateResource, context *wiringplugin.Wiring
 	if err != nil {
 		return nil, err
 	}
-	ingressResourceReferenceName, err := getReferenceNameByGVK(ingressDependency.SmithResources, schema.GroupVersionKind{
-		Group:   "extensions",
-		Version: "v1beta1",
-		Kind:    k8s.IngressKind,
-	})
-	if err != nil {
-		return nil, err
+	shape, found := ingressDependency.Contract.FindShape(knownshapes.IngressEndpointShape)
+	if !found {
+		return nil, errors.Errorf("shape %q is required to create binding for %q but was not found", knownshapes.IngressEndpointShape, ingressDependency.Name)
 	}
-	referenceName := wiringutil.ReferenceName(ingressResourceReferenceName, kubeIngressRefMetadata, kubeIngressRefMetadataEndpoint)
-	references = append(references, smith_v1.Reference{
-		Name:     referenceName,
-		Resource: ingressResourceReferenceName,
-		Path:     kubeIngressRefMetadataEndpointPath,
-		Example:  kubeIngressRefExample,
-	})
+	ingressEndpoint := shape.(*knownshapes.IngressEndpoint).Data.IngressEndpoint
```

Comment:
check if cast succeeded just in case

Reply:
Don't bother, we'll use a special conversion function in these places. I'm working on it in a branch. We'll need to handle unstructured->typed shape conversion in such places.

Reply:
https://github.com/atlassian/voyager/pull/43